### ISSUE TITLE
Connecting base

### ISSUE THREAD

**Issue by nickcoutsos** [nickcoutsos/dactyl-flatpacked#3]
> a base that connects each half of the project together would add significant cost, but would greatly positively impact the sturdiness of the build in the long run, and provide a space to keep the wiring, microcontrollers, and ports etc

This is a good idea as well. Thin pieces can be used just to connect the thumb and finger cluster. Multiple pieces can be attached at different heights to reduce some of the stresses on the acrylic. Alternatively, one big piece that covers the entire footprint should be relatively strong if you're comfortable with the cost.

Speaking of which, this could be an area where people have different preferences so I can see there being value in generating a template for the footprint in addition to a couple of connector options. This way there's room for people to make their own customization in any old vector editor.

**Comment by 7t2** [comment]
Yes, I was even thinking of making the base out of thicker wood or some other material like that. So having the base seperate to the rest of the build might actually be a pretty good idea. Adding a second layer that supports the acrylic beams would be a good idea too, and wouldn't need to be as big as the bottom layer, just surrounding the thumb cluster supports.

Another benefit of having an acrylic base included is that you can slot the pieces into it while you glue and assemble the rest of it together, only gluing it to the base once the wiring has been tested. This would be especially useful for the thumb cluster, because it's difficult making sure the legs are parallel and at the right angle all at once.

Or instead of gluing, one could drill holes in the base and the pieces touching the base, so that they could be wired together, providing a strong but removable method of joining the pieces together.

So maybe having two different material optimised SVG's would be best, one with the acrylic bases, and one without. Because I would really enjoy and use the acrylic base, as I'm sure many others would (it's expensive to have the tools necessary to build a nice wooden base), and it would be much more affordable to have them all together in one file to print.

**Comment by nickcoutsos** [comment]
I did some work on a base piece a week or two ago and just realized I never shared the progress. This builds on the "joint" idea, cutting out the lower corners of the thumb cross supports so that they can slot into something.

![base-cutout](https://user-images.githubusercontent.com/1165066/168496670-edc6aae6-5f9f-4fb5-899b-8d002f6638a2.png)

The footprint is based on the innermost and outermost columns for the thumb and finger clusters, and uses some bezier curves for aesthetics. The larger cutouts are to minimize material cost but won't be useful for soldering repairs Might be better to use hot glue or another temporary adhesive on the base. Nesting can be used to cut other pieces out of that space, but I may still come up with a variation that doesn't cover the full length of either cluster.

![base-and-brace](https://user-images.githubusercontent.com/1165066/168496776-b702bcc2-876e-4d10-9651-ffd0b4c407a7.png)
A smaller brace piece is used to connect the finger and thumb clusters at a higher point so that the weight of one half won't snap the base when picked up by the other end.

Here's how the cross supports fit in
![brace-cross-supports](https://user-images.githubusercontent.com/1165066/168496871-f3707472-c4a6-4e91-9173-e2ed30fc26c7.png)
One key difference between this and the original design is how the back cross support extends across the entire thumb cluster in front of the finger cluster's front cross support instead of spanning two thumb columns behind it. The result is that there isn't a convenient place to snake wires connecting the thumb cluster. I'll probably need to leave a cutout in one of the rear cross supports.

The other thing I wanted to get out of the brace is a way to mount the controller board.
![promicro-placement](https://user-images.githubusercontent.com/1165066/168496900-2b051c4f-dd83-4040-803b-764bff016e97.png)

I thought I'd have more room, but I can't bring the brace up too high do to the finger column supports. I'll figure something out, but I guess I need to account for a TRRS connector for people who would use it with QMK anyway, so there will likely be a third piece sitting across the base and brace pieces. I guess another option is to leave more room on the base piece and lay the electronics down flat (or stacked maybe, since people may need room for a battery). I'm partial to the sideways thing, though, so I'm going to keep thinking about my options.

**Comment by 7t2** [comment]
Bro, your design for the base and the brace piece is awesome! I'm so excited to see what you've done, it's super functional and looks really damn good! Having the hole in the base to save material by cutting other pieces inside it is a super clever idea, and the brace along with the base completely solves all of the thumb cluster stability issues. Cutting a hole to snake wires would be convenient, and using the brace to mount the controller board like that is a stroke of genius. Keep it up man, you've got me excited and inspired
